packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2762,25 +2762,27 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         summaryLogger: ITelemetryLogger,
     ) {
         const readAndParseBlob = async <T>(id: string) => readAndParse<T>(this.storage, id);
-        const { snapshotTree } = await this.fetchSnapshotFromStorage(
-            ackHandle,
-            summaryLogger,
-            {
-                eventName: "RefreshLatestSummaryGetSnapshot",
-                ackHandle,
-                summaryRefSeq,
-                fetchLatest: false,
-            },
-        );
-        const result = await this.summarizerNode.refreshLatestSummary(
-            proposalHandle,
-            summaryRefSeq,
-            async () => snapshotTree,
-            readAndParseBlob,
-            summaryLogger,
-        );
-
-        // Notify the garbage collector so it can update its latest summary state.
+        const snapshotTreeFetcher = async () => {
```

Comment:
Can you please add a comment that we should not fetch the snapshot here as its very expensive and is not always needed. It should only be done in the summarizerNode if required.
This would avoid any such regressions in the future.